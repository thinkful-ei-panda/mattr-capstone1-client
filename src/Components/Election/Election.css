
p.light{
    color: #243b55;
}

.election-img{
    width: 50%;
}

@media screen and (min-width: 1200px) {
.scorecard-entry {
    margin: 20px;
    padding: 10px;
    width: 60%;
    background-color: #03e9f4;
    box-shadow: 2px 2px 12px 3px rgba(255, 255, 255, 0.76);
    }

    .election-img{
        width: 100%;
    }

}