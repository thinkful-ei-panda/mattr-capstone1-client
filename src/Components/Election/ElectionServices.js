import config from "../../config"

const ElectionServices = {

   getElection1Data(){
    return  fetch(`${config.API_ENDPOINT}/election/1`, {
      headers:{}
    })
    .then(res => res.json() )
  },

  getCandidate1Data(){
    return fetch(`${config.API_ENDPOINT}/candidate/1`)
    .then(res => {
      console.log(res)
     return res.json()
    })
   
  },

  getCandidate2Data(){
    fetch(`${config.API_ENDPOINT}/candidate/2`)
    .then(res => {
     return res.json()
    })
    .then(res => {
      console.log(res)
      this.setState({candidate2: res})
    })
  },

  getAllVoteData(){
    fetch(`${config.API_ENDPOINT}/vote`)
    .then(res => {
     return res.json()
    })
    .then(res => {
      console.log(res)
      this.setState({votes: res})
    })
  }
}

export default ElectionServices