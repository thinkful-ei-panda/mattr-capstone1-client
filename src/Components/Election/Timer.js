import React, { Component } from 'react'

const moment = require('moment')

export default class Timer extends Component {

//     state = { election:[] }
//   componentDidMount(){
//     fetch("http://localhost:8000/api/election/1")
//     .then(res => {
//      return res.json()
//     })
//     .then(res => {
//       console.log(res)
//       this.setState({election: res})
//     })
//   }

    render() {
        const { election } = this.props
        const electionDay = moment("20201104", "YYYYMMDD").fromNow()
        return (
            <div className='momentum'>
               <h4>Election Day Deadline: {electionDay} </h4> 

                
            </div>
        )
    }
}
