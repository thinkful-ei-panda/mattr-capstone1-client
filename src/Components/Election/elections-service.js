const xss = require('xss')

const ElectionsService = {
  getAllElections(db) {
    return db
      .from('poll_data_election AS election')
      .select(
        'election.id',
        'election.election_name',
        'election.date_created',
        'election.date_end',
        db.raw(
          `count(DISTINCT vote) AS number_of_votes`
        ),
        db.raw(
          `json_strip_nulls(
            json_build_object(
              'election_id', electio.id,
              'candidate_id', candidate.candidate_id,
              'user_id', user.user_id,
              'date_created', user.date_created,
            )
          ) AS "voter"`
        ),
      )
      .leftJoin(
        'poll_data_vote AS vote',
        'election.id',
        'vote.election_id',
      )
      .leftJoin(
        'poll_data_user AS user',
        'election.election_id',
        'user.user_id',
      )
      .leftJoin(
        'poll_data_candidate AS candidate',
        'candidate.candidate_id',
        'candidate.candidate_name',
        
      )
      .groupBy('election.id', 'candidate.id')
  },

  getById(db, id) {
    return ElectionsService.getAllArticles(db)
      .where('election.id', id)
      .first()
  },

  getVotesForElection(db, election_id) {
    return db
      .from('poll_data_vote AS vote')
      .select(
        'vote.id',
        'vote.date_created',
        db.raw(
          `json_strip_nulls(
            row_to_json(
              (SELECT tmp FROM (
                SELECT
                  user.id,
                  user.date_created,
                  user.date_modified
              ) tmp)
            )
          ) AS "user"`
        )
      )
      .where('vote.election_id', election_id)
      .leftJoin(
        'poll_data_user AS user',
        'vote.user_id',
        'user.id',
      )
      .groupBy('vote.id', 'user.id')
  },

  serializeArticle(election) {
    const { voter } = election
    return {
      id: election.id,
      style: election.style,
      title: xss(election.title),
      content: xss(election.content),
      date_created: new Date(election.date_created),
      number_of_comments: Number(election.number_of_comments) || 0,
      voter: {
        id: voter.id,
        date_created: new Date(voter.date_created),
        date_modified: new Date(voter.date_modified) || null
      },
    }
  },

  serializeArticleComment(vote) {
    const { user } = vote
    return {
      id: vote.id,
      article_id: vote.article_id,
      text: xss(vote.text),
      date_created: new Date(vote.date_created),
      user: {
        id: user.id,
        date_created: new Date(user.date_created),
        date_modified: new Date(user.date_modified) || null
      },
    }
  },
}

module.exports = ElectionsService
