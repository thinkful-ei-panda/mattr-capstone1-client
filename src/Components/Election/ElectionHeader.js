import React, { Component } from 'react'
import ElectionServices from './ElectionServices'
const moment = require('moment')


export default class ElectionHeader extends Component {
    state = {
        election:{}
    }
    componentDidMount(){
        ElectionServices.getElection1Data().then((election)=>this.setState({election}))
    }
    
    render() {
    
        return (
            <>
                <h2>{ this.state.election.election_name }</h2>
                <h3> Election Day Deadline: { moment(this.state.election.date_end).fromNow() } </h3>
            </>
        )
    }
}
