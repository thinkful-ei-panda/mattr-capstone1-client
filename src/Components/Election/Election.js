import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import config from '../../config'
import Democrat from '../Images/joe-biden.jpg'
import Republican from '../Images/donald-trump.jpeg'
import ElectionHeader from './ElectionHeader'
import Nav from '../Nav'

export default class Election extends Component {

  constructor(props) {
    super(props);
    this.state = {
      election1: [],
      candidate1: [],
      candidate2: [],
      votes: [],
    }
  }
 
  getCandidate1Data(){
    fetch(`${config.API_ENDPOINT}/candidate/1`)
    .then(res => {
     return res.json()
    })
    .then(res => {
      console.log(res)
      this.setState({candidate1: res})
    })
  }

  getCandidate2Data(){
    fetch(`${config.API_ENDPOINT}/candidate/2`)
    .then(res => {
     return res.json()
    })
    .then(res => {
      console.log(res)
      this.setState({candidate2: res})
    })
  }

  getAllVoteData(){
    fetch(`${config.API_ENDPOINT}/vote`)
    .then(res => {
     return res.json()
    })
    .then(res => {
      console.log(res)
      this.setState({votes: res})
    })
  }

  componentDidMount(){
    this.getCandidate1Data()
    this.getCandidate2Data()
    this.getAllVoteData()
  }

    render() {
     function countTotalVotes(arr){
            return arr.length
          }
        
      function countVotes(arr, num){
        let results = 0;
        for(let i in arr){
          if(arr[i].candidate_id === num){
            results += 1
          } 
        }
        return results
      }


      const totalVotes = countTotalVotes(this.state.votes)
      const bidenVotes = countVotes(this.state.votes, this.state.candidate1.candidate_id)
      const trumpVotes = countVotes(this.state.votes, this.state.candidate2.candidate_id)
        return (
            <section className='election-card'>
              <Nav />  
              <ElectionHeader />
        <h3>Total Votes: {totalVotes}</h3>

              <div className='scorecard-entry'>
                  <img src={Democrat} alt="Joe Biden" />
                  <div className="stats">
                  <h4 className='light'>{this.state.candidate1.candidate_name}</h4>
                  <p className='light'>Votes: {bidenVotes} </p>
                  </div>
              </div>

              <div className='scorecard-entry'>
                  <img src={Republican} alt="Donald Trump" />
                  <div className="stats">
                  <h4 className='light' >{this.state.candidate2.candidate_name}</h4>
                  <p className='light'>Votes: {trumpVotes} </p>
                  </div>
              </div>
              
              <Link to='/Vote' className='vote-link'>Vote</Link>
          </section>
        )
    }
  }

