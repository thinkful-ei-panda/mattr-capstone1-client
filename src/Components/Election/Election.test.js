import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Election from './Election';
describe(`Election Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <Election />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <Election />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})