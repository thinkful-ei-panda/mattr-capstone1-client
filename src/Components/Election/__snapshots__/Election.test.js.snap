// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Election Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="wrapper"
>
  <div
    className="Nav"
  >
    <a
      href="/Home"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/Election"
      onClick={[Function]}
    >
      Elections
    </a>
    <a
      href="/RegistrationForm"
      onClick={[Function]}
    >
       
      Register
       
    </a>
    <a
      href="/Login"
      onClick={[Function]}
    >
      Login
    </a>
  </div>
  <div
    className="login-box"
  >
    <h2 />
    <h3>
       Election Day Deadline: 
      a few seconds ago
       
    </h3>
    <h3>
      Total Votes: 
      0
    </h3>
    <div
      className="election-cards"
    >
      <div
        className="scorecard-entry"
      >
        <img
          alt="Joe Biden"
          src="joe-biden.jpg"
        />
        <div
          className="stats"
        >
          <h4
            className="light"
          />
          <p
            className="light"
          >
            Votes: 
            0
             
          </p>
        </div>
      </div>
      <div
        className="scorecard-entry"
      >
        <img
          alt="Donald Trump"
          src="donald-trump.jpeg"
        />
        <div
          className="stats"
        >
          <h4
            className="light"
          />
          <p
            className="light"
          >
            Votes: 
            0
             
          </p>
        </div>
      </div>
    </div>
    <a
      className="vote-link"
      href="/Vote"
      onClick={[Function]}
    >
      Vote
    </a>
  </div>
</div>
`;
